### PR TITLE
some documentation changes :bookmark:

### PR DESCRIPTION
**Pull request by eliyabar**
1.updating Preevy actions, up and down docs.
2.fix cli reference to be spread, command by command. 3.add automation for new cli reference pages.
4.add azure documentation
5.fix some typos :pencil2:

### CODE REVIEW

File: site/docs/ci/example-github-actions.md [hunk 1]
```diff
@@ -5,7 +5,9 @@ title: "Example: Github Actions"
 
 # Github Actions
 
-In this section we'll show an example of how to run Preevy in GitHub Actions pipeline.
+In this section we'll show an example of how to run Preevy using our GitHub Actions [preevy-up](https://github.com/marketplace/actions/preevy-up) and [preevy-down](https://github.com/marketplace/actions/preevy-down).
```
Comment (royra) [inline site/docs/ci/example-github-actions.md:8]: probably need to explain it's using AWS lightsail and S3 specifically (and that is the reason for the authentication)

File: site/docs/ci/example-github-actions.md [hunk 2]
```diff
@@ -16,41 +18,27 @@ Once configured, use the [AWS for GitHub Actions](https://github.com/marketplace
...
 
-And load the Preevy profile you [configured earlier](/ci/overview#how-to-run-preevy-from-the-ci), e.g.
-
-```bash
-preevy init --from "s3://preview-8450209857-ci?region=us-east-1"
-```
-
-Now run `preevy up` from the directory that contains your `docker-compose.yml` file
+Now, using Preevy-UP GitHub action arg `docker-compose-yaml-paths` from the directory that contains your `docker-compose.yml` file.
+if you have multiple docker compose files, you can add them as a comma seperated string like so `'docker-compose.yml,docker-compose.dev.yml'`
```
Comment (royra) [inline site/docs/ci/example-github-actions.md:41]: this belongs in an inline comment at the `docker-compose-yaml-paths` arg

File: site/docs/ci/example-github-actions.md [hunk 3]
```diff
@@ -16,41 +18,27 @@ Once configured, use the [AWS for GitHub Actions](https://github.com/marketplace
...
 ```
 
-And load the Preevy profile you [configured earlier](/ci/overview#how-to-run-preevy-from-the-ci), e.g.
-
-```bash
-preevy init --from "s3://preview-8450209857-ci?region=us-east-1"
-```
-
-Now run `preevy up` from the directory that contains your `docker-compose.yml` file
+Now, using Preevy-UP GitHub action arg `docker-compose-yaml-paths` from the directory that contains your `docker-compose.yml` file.
```
Comment (royra) [inline site/docs/ci/example-github-actions.md:40]: nit: Preevy-UP is weird styling. I would go with `preevy-up` or "the Preevy Up"

File: site/docs/ci/example-github-actions.md [hunk 4]
```diff
@@ -64,24 +52,71 @@ on:
 permissions:
   id-token: write
   contents: read
+  pull-requests: write
```
Comment (royra) [inline site/docs/ci/example-github-actions.md:55]: explain the reasons for the permissions

File: site/docs/ci/example-github-actions.md [hunk 5]
```diff
@@ -64,24 +52,71 @@ on:
...
+        with:
+          profile-url: "s3://preevy-12345678-my-profile?region=eu-west-1"
+          docker-compose-yaml-paths: "./docker/docker-compose.yaml"
+      - uses: mshick/add-pr-comment@v2
+        with:
+          message: ${{ steps.preevy.outputs.urls-markdown }} 
+```
+
+# Preevy-Down
+Use this action to stop and delete a preview environment using the Preevy CLI.
```
Comment (royra) [inline site/docs/ci/example-github-actions.md:76]:
```suggestion
Use this action to stop and delete a preview environment using the Preevy CLI when the Pull Request is merged or closed.
```

File: site/docs/ci/example-github-actions.md [hunk 6]
```diff
@@ -64,24 +52,71 @@ on:
...
+
+## Teardown Preevy
```
Comment (royra) [inline site/docs/ci/example-github-actions.md:78]:
```suggestion
## Teardown the Preevy environment
```

File: site/docs/ci/example-github-actions.md [hunk 7]
```diff
@@ -16,41 +18,27 @@ Once configured, use the [AWS for GitHub Actions](https://github.com/marketplace
...
+## Running Preevy
+Make sure your code is checked out before using the preevy up action, using the [Checkout](https://github.com/marketplace/actions/checkout) action:
 
 ```yaml
 - uses: actions/checkout@v3
+- uses: livecycle/preevy-up-action@latest
 ```
 
-Now you can install the Preevy CLI:
+With the `profile-url` arg, load the Preevy profile you [configured earlier](/ci/overview#how-to-run-preevy-from-the-ci), e.g.
```
Comment (royra) [inline site/docs/ci/example-github-actions.md:34]: explain it's using S3 with permissions

File: site/docs/drivers/azure.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
+---
+sidebar_position: 3
+title: Microsoft Azure Driver
+---
+
+# Microsoft Azure Driver
+
+Preevy can provision virtual machines on Microsoft Azure using the `azure` driver.
+Microsoft Azure also offer free 12 months for new users which is suited for trying out preevy.
```
Comment (royra) [inline site/docs/drivers/azure.md:9]:
```suggestion
Microsoft Azure also offers free 12 months for new users which is suited for trying out preevy.
```

File: site/docs/drivers/azure.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+### Supported flags
+- `--azure-region` - Microsoft Azure region in which resources will be provisioned
+- `--azure-resource-group-name` - Microsoft Azure resource group name
+- `--azure-subscription-id` - Microsoft Azure subscription id
+- `--azure-vm-size` - Machine type to be provisioned, defaults to `Standard_B2s`
```
Comment (royra) [inline site/docs/drivers/azure.md:15]: this is optional, right?

File: site/docs/drivers/azure.md [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+
+### Credentials Configuration
+Preevy uses the Microsoft Azure SDK which uses the `@azure/identity` package to get the application default credentials (https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/identity/identity#defaultazurecredential).
```
Comment (royra) [inline site/docs/drivers/azure.md:18]:
```suggestion
Preevy uses the Microsoft Azure SDK which can obtain the application [default credentials](https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/identity/identity#defaultazurecredential).
```

File: site/docs/drivers/gcp-gce.md [hunk 11]
```diff
@@ -1,12 +1,12 @@
...
-sidebar_position: 1
+sidebar_position: 2
 title: GCP Compute Engine Driver
 ---
 
 # GCP Compute Engine Driver
 
 Preevy can provision virtual machines on GCP using the `gce` driver.
-Google compute engine provisioning time for a VM is usually less then a minute, the default machine size in use is e2-small (2GB, 2vCpu) which cost around $12 per month.
+Google compute engine provisioning time for a VM is usually less than a minute, the default machine size in use is e2-small (2GB, 2vCpu) which cost around $12 per month.
```
Comment (royra) [inline site/docs/drivers/gcp-gce.md:9]:
```suggestion
Google compute engine provisioning time for a VM is usually less than a minute, the default machine size in use is e2-small (2GB, 2vCpu) which costs around $12 per month.
```

**Review by royra — APPROVED** [review]
see minor comments

File: packages/cli/scripts/prepare_and_copy_docs.sh [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+folder_path="$parent_folder/../docs"
+
+# Define the relative paths
+docs_source_path="../docs"
+commands_destination_path="../../../site/docs/cli-reference/commands/"
+readme_source_path="../README.md"
+index_destination_path="../../../site/docs/cli-reference/index.md"
+
+# Find and replace URLs containing /dist/commands/ with /packages/cli/src/commands/
+find "$parent_folder/$docs_source_path" -type f -exec sed -i.bak 's#/dist/commands/#/packages/cli/src/commands/#g' {} +
```
[the remark this answers is not in the document]
Comment (eliyabar) [reply packages/cli/scripts/prepare_and_copy_docs.sh:16]: but when i do this i get an error that sed -i isn't supported for some reason.

File: site/docs/drivers/gcp-gce.md (hunk 11 above)
Comment (eliyabar) [reply site/docs/drivers/gcp-gce.md:9]: touché

**Pull request merged**